TASVideos/Services/WikiToTextRenderer.cs

```diff
@@ -53,10 +66,34 @@ public bool CheckCondition(string condition)
 				return result;
 			}
 
-			public Task RunViewComponentAsync(TextWriter w, string name, IReadOnlyDictionary<string, string> pp)
+			public async Task RunViewComponentAsync(TextWriter w, string name, IReadOnlyDictionary<string, string> pp)
 			{
-				// TODO: This needs to be handled in concert with Wiki changes
-				return Task.CompletedTask;
+				var componentExists = ModuleParamHelpers.TextComponents.TryGetValue(name, out Type? textComponent);
+				if (!componentExists)
+				{
+					return;
+				}
+
+				var invokeMethod = textComponent!.GetMethod("RenderTextAsync");
+
+				if (invokeMethod == null && textComponent.GetMethod("RenderText") != null)
+				{
+					throw new NotImplementedException("Sync method not supported yet");
+				}
+
+				if (invokeMethod == null)
+				{
+					throw new InvalidOperationException($"Could not find an RenderText method on ViewComponent {textComponent}");
+				}
+
+				var paramObject = ModuleParamHelpers
+					.GetParameterData(w, name, invokeMethod, _wikiPage, pp);
+
+				var module = _serviceProvider.GetRequiredService(textComponent);
+				var task = (Task)invokeMethod.Invoke(module, paramObject.Values.ToArray())!;
```

Comment:
You should be able to just cast to `Task<string>` here and leave out the reflection bit below.